From 3d5024fb9c904a649d07f0def3a90b3d36611215 Mon Sep 17 00:00:00 2001
From: "Dr. David Alan Gilbert" <dgilbert@redhat.com>
Date: Wed, 12 Jan 2022 13:21:57 +0000
Subject: [PATCH 2/3] x86: Add q35 RHEL 9.0.0 machine type

RH-Author: Dr. David Alan Gilbert <dgilbert@redhat.com>
RH-MergeRequest: 61: x86: Add rhel 8.6.0 & 9.0.0 machine types
RH-Commit: [2/2] 743378502459b978efd632271f97ddb824422203 (dagrh/c-9-s-qemu-kvm)
RH-Bugzilla: 1945666
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>
RH-Acked-by: Stefano Garzarella <sgarzare@redhat.com>
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>

Add a rhel-9.0.0 q35 machine type; it's currently identical to 8.6.0;
but having a separate machine type will make life easier in the future
when the 8.x types go away.

Note: The smbios stream product name has now changed to 'RHEL'

bz: https://bugzilla.redhat.com/show_bug.cgi?id=1945666
Signed-off-by: Dr. David Alan Gilbert <dgilbert@redhat.com>
---
 hw/i386/pc_q35.c | 21 +++++++++++++++++++--
 1 file changed, 19 insertions(+), 2 deletions(-)

diff --git a/hw/i386/pc_q35.c b/hw/i386/pc_q35.c
index 0c25305f15..bf9ad32f0e 100644
--- a/hw/i386/pc_q35.c
+++ b/hw/i386/pc_q35.c
@@ -646,6 +646,23 @@ static void pc_q35_machine_rhel_options(MachineClass *m)
     compat_props_add(m->compat_props, pc_rhel_compat, pc_rhel_compat_len);
 }
 
+static void pc_q35_init_rhel900(MachineState *machine)
+{
+    pc_q35_init(machine);
+}
+
+static void pc_q35_machine_rhel900_options(MachineClass *m)
+{
+    PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
+    pc_q35_machine_rhel_options(m);
+    m->desc = "RHEL-9.0.0 PC (Q35 + ICH9, 2009)";
+    pcmc->smbios_stream_product = "RHEL";
+    pcmc->smbios_stream_version = "9.0.0";
+}
+
+DEFINE_PC_MACHINE(q35_rhel900, "pc-q35-rhel9.0.0", pc_q35_init_rhel900,
+                  pc_q35_machine_rhel900_options);
+
 static void pc_q35_init_rhel860(MachineState *machine)
 {
     pc_q35_init(machine);
@@ -654,8 +671,9 @@ static void pc_q35_init_rhel860(MachineState *machine)
 static void pc_q35_machine_rhel860_options(MachineClass *m)
 {
     PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
-    pc_q35_machine_rhel_options(m);
+    pc_q35_machine_rhel900_options(m);
     m->desc = "RHEL-8.6.0 PC (Q35 + ICH9, 2009)";
+    m->alias = NULL;
     pcmc->smbios_stream_product = "RHEL-AV";
     pcmc->smbios_stream_version = "8.6.0";
 }
@@ -674,7 +692,6 @@ static void pc_q35_machine_rhel850_options(MachineClass *m)
     PCMachineClass *pcmc = PC_MACHINE_CLASS(m);
     pc_q35_machine_rhel860_options(m);
     m->desc = "RHEL-8.5.0 PC (Q35 + ICH9, 2009)";
-    m->alias = NULL;
     pcmc->smbios_stream_product = "RHEL-AV";
     pcmc->smbios_stream_version = "8.5.0";
     compat_props_add(m->compat_props, hw_compat_rhel_8_5,
-- 
2.27.0

